### PR TITLE
feat(gameservices): generate library

### LINKED ISSUE #8020: Game Services API: `gameservices.googleapis.com`

### PR DESCRIPTION
**Pull request by scotthart**
fixes #8020

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021)

### CODE REVIEW

File: external/googleapis/update_libraries.sh [hunk 1]
```diff
@@ -69,6 +69,7 @@ declare -A -r LIBRARIES=(
       "@com_google_googleapis//google/cloud/gkehub/v1/multiclusteringress:multiclusteringres_cc_grpc" \
       "@com_google_googleapis//google/cloud/gkehub/v1/configmanagement:configmanagement_cc_grpc"
   )"
+  ["gameservices"]="@com_google_googleapis//google/cloud/gaming/v1:gaming_cc_grpc"
```
Comment (dbolduc) [inline external/googleapis/update_libraries.sh:72]: I think this should go above `gkehub`

File: generator/generator_config.textproto [hunk 2]
```diff
@@ -176,6 +176,35 @@ service {
   retryable_status_codes: ["kUnavailable"]
 }
 
+# Game Services
```
Comment (dbolduc) [inline generator/generator_config.textproto:179]: same nit about moving this above `GKE Hub`

File: BUILD.bazel [hunk 3]
```diff
@@ -213,6 +213,22 @@ cc_library(
...
 
+cc_library(
+    name = "experimental-gameservices",
+    deps = [
+        "//google/cloud/gameservices:google_cloud_cpp_gameservices",
+    ],
+)
+
+cc_library(
+    name = "experimental-gameservices-mocks",
```
Comment (dbolduc) [inline BUILD.bazel:224]: s/-mocks/_mocks/

File: BUILD.bazel [hunk 4]
```diff
@@ -213,6 +213,22 @@ cc_library(
...
+    deprecation = "please use //:gameservices_mocks instead.",
+    tags = ["manual"],
```
Comment (dbolduc) [inline BUILD.bazel:226]: let's not do this

File: google/cloud/gameservices/quickstart/quickstart.cc [hunk 5]
```diff
@@ -12,23 +12,23 @@
...
   namespace gameservices = ::google::cloud::gameservices;
-  auto client =
-      gameservices::Client(gameservices::MakeConnection(/* EDIT HERE */));
+  auto client = gameservices::GameServerClustersServiceClient(
+      gameservices::MakeGameServerClustersServiceConnection());
 
-  auto const project = google::cloud::Project(argv[1]);
-  for (auto r : client.List /*EDIT HERE*/ (project.FullName())) {
+  auto const realm = "projects/" + std::string(argv[1]) + "/locations/" +
+                     std::string(argv[2]) + "/realms/" + std::string(argv[3]);
```
Comment (dbolduc) [inline google/cloud/gameservices/quickstart/quickstart.cc:30]: optional consistency nit: I think the non-Brad quickstarts would have said: 
```cc
std::string{"projects/"} + argv[1] + "/locations/" + argv[2] + "/realms/" + argv[3];
```

**Review by dbolduc — CHANGES_REQUESTED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021)* status: 0 of 104 files reviewed, 5 unresolved discussions (waiting on @dbolduc)

___
*[BUILD.bazel, line 224 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021#-MtoMbhPDweV16FTs_Jq-r2-224:-MtoMiqeCoLmV9ESofpL:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c5785e1ce904081bc5aab85658b0fca2d59ee991/BUILD.bazel#L224)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

s/-mocks/_mocks/
</blockquote></details>

Done.
___
*[BUILD.bazel, line 226 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021#-MtoMbhPDweV16FTs_Jq-r2-226:-MtoMjPdBXQ8h0WUhD8b:b-ictfve) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c5785e1ce904081bc5aab85658b0fca2d59ee991/BUILD.bazel#L226)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

let's not do this
</blockquote></details>

copy paste monkey
___
*[external/googleapis/update_libraries.sh, line 72 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021#-MtnAD8fYuvGUTPuoXmn-r1-72:-MtoMt5n7moj-oWp2Hj0:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/439897f7f0a71cb69a262962a9538c7886c1d772/external/googleapis/update_libraries.sh#L72)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

I think this should go above `gkehub`
</blockquote></details>

Done.
___
*[generator/generator_config.textproto, line 179 at r1](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021#-MtnAD8fYuvGUTPuoXmo-r1-179:-MtoMu646GYkVWLvOni4:b-896fix) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/439897f7f0a71cb69a262962a9538c7886c1d772/generator/generator_config.textproto#L179)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

same nit about moving this above `GKE Hub`
</blockquote></details>

Done.
___
*[google/cloud/gameservices/quickstart/quickstart.cc, line 30 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8021#-MtnAD8hHGjG0EG6WtfC-r2-30:-MtoN1hIDC4VZnqNB_3G:b-1ralho) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/c5785e1ce904081bc5aab85658b0fca2d59ee991/google/cloud/gameservices/quickstart/quickstart.cc#L30)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

optional consistency nit: I think the non-Brad quickstarts would have said: 
```cc
std::string{"projects/"} + argv[1] + "/locations/" + argv[2] + "/realms/" + argv[3];
```
</blockquote></details>

It's consistent with the others I've done like this. If there's a group consensus on what we prefer, they can all be modified later.

File: BUILD.bazel [hunk 6]
```diff
@@ -221,9 +221,8 @@ cc_library(
 )
 
 cc_library(
-    name = "experimental-gameservices-mocks",
+    name = "experimental-gameservices_mocks",
     deprecation = "please use //:gameservices_mocks instead.",
```
Comment (dbolduc) [inline BUILD.bazel:225]: 🐒

**Review by dbolduc — APPROVED** [review]
I disabled auto merge, because the mocks library is still marked as deprecated.

**Pull request merged**